hack/helm-reference-gen/doc_node.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// DocNode is a node in the final generated reference document.
+// For example this would be a single DocNode:
+// ```
+// - `global` ((#v-global)) - Holds values that affect multiple components of the chart.
+// ```
+type DocNode struct {
+	// Column is the character column (i.e. indent) this node should be displayed
+	// at.
+	// For example, if this is a root node, then its column will be 0 because it
+	// shouldn't be indented.
+	Column int
+
+	// ParentBreadcrumb is the path to this node's parent from the root.
+	// It is used for the HTML anchor, e.g. `#v-global-name`.
+	// If this node were global.name, then this would be set to "global".
+	ParentBreadcrumb string
+
+	// ParentWasMap is true when the parent of this node was a map.
+	ParentWasMap bool
+
+	// Key is the key of this node, e.g. if `key: value` then Key would be "key".
+	Key string
+
+	// Default is the default value for this node, e.g. if key defaults to false,
+	// Default would be "false".
+	Default string
+
+	// Comment is the YAML comment that described this node.
+	Comment string
+
+	// KindTag is the YAML parsed kind tag from the YAML library. This has values
+	// like "!!seq" and "!!str".
+	KindTag string
+
+	// Children are other nodes that should be displayed as sub-keys of this node.
+	Children []DocNode
+}
+
+// Validate returns an error if this node is invalid, else nil.
+func (n DocNode) Validate() error {
+	if strings.Contains(n.Kind(), "unknown") {
```

Comment:
completely cosmetic comment :  searching for `unknown` instead of the string we add in `Kind()` :  `unknown kind`.
If you do actually address it should we make the string a const?

Reply:
yeah good catch! will do